DoSomething/northstar
# drupal password checker

#### What does this PR do?
adds DrupalPasswordChecker, Password, and Registrar services to log in a user in with drupal password if there is no password. It then updates the users by saving the password to `$user->password` and deletes `$user->drupal_password`.

#### Where should the reviewer start? 
AuthController -> Registrar -> DrupalPasswordChecker -> Password

#### How should this be manually tested?
1. Find a user that only has a `drupal_password` (or delete a user's `password` by going into mongo and running `db.users.update({email: "user_email@email.com"}), {$unset: {password:1}})` in the 
userapi db). 

2. In Postman/Paw, hit the `/login` endpoint. Body should have `email` and `password`: 
```
{
  "email": "user_email@email.com", 
  "password": "<drupal-pw>"
}
```

3. Even though the user did not have a `password`, it should log the user in with the `drupal_password` and return the user object like so: 
```
{
  "data": {
    "_id": "563d251cbffebc482d8b4568",
    "email": "clee@dosomething.org",
    "drupal_id": "1704953",
    "birthdate": "1995-04-12 00:00:00",
    "first_name": "Chloe",
    "country": "US",
    "updated_at": "2015-11-12 21:06:57",
    "created_at": "2015-11-06 22:09:32",
    "session_token": "c2dIdWE2UEF0SzBNNTZsSEN5TzI1bmdSSDBKTEQxb3p0eGk5R2c0Rk9VPQ=="
  }
}
```
4. Check the user in mongo and make sure that `drupal_password` = `null` and there is now a stored `password` in the user object by using `db.users.find({email: "user_email@email.com"}).pretty()`

#### What are the relevant tickets?
Fixes #226
